Classes/Hooks/KeSearchIndexer.php

```diff
@@ -70,13 +70,35 @@ public function customIndexer(&$indexerConfig, &$indexerObject)
             $abstract = strip_tags($originalObject->getKeSearchAbstract($index));
             $content = strip_tags($originalObject->getKeSearchContent($index));
             $fullContent = $title . "\n" . $abstract . "\n" . $content;
+            $tags = [];
+            foreach ($originalObject->getCategories() as $category) {
+                $tags[] = "#syscat{$category->getUid()}#";
+            }
+            $tags = implode(',', $tags);
 
             // @todo Add year and month information
-            $additionalFields = [];
+            $additionalFields = [
+                'sortdate' => $index->getStartDateComplete()->getTimestamp(),
+                'orig_uid' => $index->getUid(),
+                'orig_pid' => $index->getPid(),
+            ];
 
-            $indexerObject->storeInIndex($indexerConfig['storagepid'], $title, 'calendarize', $indexerConfig['targetpid'],
-                $fullContent, '', '&tx_calendarize_calendar[index]=' . $index->getUid(), $abstract, 0, 0, 0, '', false,
-                $additionalFields);
+            $indexerObject->storeInIndex(
+                $indexerConfig['storagepid'],
+                $title,
+                'calendarize',
+                $indexerConfig['targetpid'],
+                $fullContent,
+                $tags,
+                "&tx_calendarize_calendar[index]={$index->getUid()}",
+                $abstract,
+                $index->_getProperty('_languageUid'),
```

Comment:
Ignore this comment. The index always has a "_languageUid". If the related Event structure do not use translations, the Index still has the sys_language_uid=0. So this line is OK :+1: